engine/runtime/src/main/scala/org/enso/compiler/pass/resolve/DocumentationComments.scala

```diff
@@ -84,6 +84,8 @@ case object DocumentationComments extends IRPass {
       case doc: IR.Comment.Documentation =>
         lastDoc = Some(doc)
         None
+      case typeAscription: IR.Type.Ascription =>
+        Some(typeAscription.asInstanceOf[T])
```

Comment:
This isn't how this should be handled. Instead of ignoring type ascriptions for the purposes of attaching docs, the documentation should be _assigned_ to the type ascription, and then during the `TypeSignatures` pass the associated documentation should be moved onto the definition itself.